entity-service/src/integrationTest/java/org/hypertrace/entity/service/service/EntityQueryServiceTest.java

```diff
@@ -0,0 +1,166 @@
+        = new org.hypertrace.entity.type.service.client.EntityTypeServiceClient(channel);
+    EntityTypeServiceClient entityTypeServiceV2Client = new EntityTypeServiceClient(channel);
+    entityTypeServiceV1Client.upsertEntityType(
+        TENANT_ID,
+        org.hypertrace.entity.type.service.v1.EntityType.newBuilder()
+            .setName(EntityType.SERVICE.name())
+            .addAttributeType(
+                AttributeType.newBuilder()
+                    .setName(EntityConstants.getValue(CommonAttribute.COMMON_ATTRIBUTE_FQN))
+                    .setValueKind(AttributeKind.TYPE_STRING)
+                    .setIdentifyingAttribute(true)
+                    .build())
+            .build());
+  }
+
+  @Test
+  public void testExecute() {
+    // create and upsert some entities
+    Entity entity1 = Entity.newBuilder()
+        .setTenantId(TENANT_ID)
+        .setEntityType(EntityType.SERVICE.name())
+        .setEntityName("Some Service 1")
+        .putIdentifyingAttributes(
+            EntityConstants.getValue(CommonAttribute.COMMON_ATTRIBUTE_FQN),
+            generateRandomUUIDAttrValue())
+        .build();
+    Entity createdEntity1 = entityDataServiceClient.upsert(entity1);
+    assertNotNull(createdEntity1);
+    assertFalse(createdEntity1.getEntityId().trim().isEmpty());
+
+    Entity entity2 = Entity.newBuilder()
+        .setTenantId(TENANT_ID)
+        .setEntityType(EntityType.SERVICE.name())
+        .setEntityName("Some Service 2")
+        .putIdentifyingAttributes(
+            EntityConstants.getValue(CommonAttribute.COMMON_ATTRIBUTE_FQN),
+            generateRandomUUIDAttrValue())
+        .build();
+    Entity createdEntity2 = entityDataServiceClient.upsert(entity2);
+
+    EntityQueryRequest queryRequest = EntityQueryRequest.newBuilder()
+        .setEntityType(EntityType.SERVICE.name())
+        .setFilter(
+            Filter.newBuilder()
+                .setOperatorValue(Operator.LT.getNumber())
+                .setLhs(Expression.newBuilder().setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName("SERVICE.createdTime").build()).build())
+                .setRhs(Expression.newBuilder().setLiteral(
+                    LiteralConstant.newBuilder().setValue(
+                        org.hypertrace.entity.query.service.v1.Value.newBuilder()
+                            .setLong(Instant.now().toEpochMilli())
+                            .setValueType(ValueType.LONG)
+                            .build())
+                        .build()).
+                    build())
+                .build())
+        .addSelection(Expression.newBuilder().setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName("SERVICE.id").build()).build())
+        .addSelection(Expression.newBuilder().setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName("SERVICE.name").build()).build())
+        .build();
+
+    // this entity will be filtered out
+    Entity entity3 = Entity.newBuilder()
```

Comment:
nit: can we add assertion that this entity created - `entity3` `assertNotNull(createdEntity3)`